platform/source/mbed_retarget.cpp

```diff
@@ -1568,6 +1568,17 @@ extern "C" {
 
 } // end of extern "C"
 
+#if defined(__MICROLIB)
+extern "C" {
+    // __aeabi_assert function definition is present in microlib boot code under
+    // Mbed OS rtos directory but this function needed in bare metal boot code so
+    // moved this function here to resolve linker issue.
+    MBED_WEAK void __aeabi_assert(const char *expr, const char *file, int line)
```

Comment:
This should probaly live in mbed_assert.c.
Also the comment should not refer to how things were done. There is a file history for that. I would remove the comment in the code. You have provided explanation in the PR summary so it is fine

Reply:
@rajkan01 Will you move this to `mbed_assert.c`?

Reply:
__aeabi_assert supplied in the ARM C library to print messages in stderr & call_abort. Either we can use the provided functionality or re-implement as explained in http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0475g/Chdhbhge.html.

on the other part, Mbed_assert is providing assert APIs to Mbed OS components so it's not the right place to mix __aeabi_assert to mbed_assert.